pkg/server/plugin/upstreamauthority/awssecret/awslib.go

```diff
@@ -34,32 +32,31 @@ func readARN(ctx context.Context, sm secretsManagerClient, arn string) (string,
 	return *resp.SecretString, nil
 }
 
-func newSecretsManagerClient(config *Configuration, region string) (secretsManagerClient, error) {
-	awsConfig := &aws.Config{
-		Region: aws.String(region),
-	}
-
-	if config.SecretAccessKey != "" && config.AccessKeyID != "" {
-		awsConfig.Credentials = credentials.NewStaticCredentials(config.AccessKeyID, config.SecretAccessKey, config.SecurityToken)
-	}
+func newSecretsManagerClient(ctx context.Context, cfg *Configuration, region string) (secretsManagerClient, error) {
+	var credsProvider aws.CredentialsProvider
+	switch {
+	case cfg.AssumeRoleARN != "":
+		stsConf := aws.Config{
```

Comment:
Looking at the example [here](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials/stscreds#hdr-Assume_Role), I think we should construct this config by loading the default config, similar to the default case.

Reply:
Fixed in latest version